iso15118/secc/states/iso15118_2_states.py

```diff
@@ -1348,51 +1346,17 @@ async def process_message(
             ac_evse_charge_params = (
                 await self.comm_session.evse_controller.get_ac_charge_params_v2()
             )
-            ev_max_voltage: Union[
-                PVEVMaxVoltageLimit, PVEVMaxVoltage
-            ] = charge_params_req.ac_ev_charge_parameter.ev_max_voltage
-            ev_max_current: Union[
-                PVEVMaxCurrentLimit, PVEVMaxCurrent
-            ] = charge_params_req.ac_ev_charge_parameter.ev_max_current
-            e_amount = charge_params_req.ac_ev_charge_parameter.e_amount
-            ev_charge_params_limits = EVChargeParamsLimits(
-                ev_max_voltage=ev_max_voltage,
-                ev_max_current=ev_max_current,
-                e_amount=e_amount,
-            )
-            departure_time = charge_params_req.ac_ev_charge_parameter.departure_time
+            ev_data_context.update_ac_charge_parameters_v2(charge_params_req.ac_ev_charge_parameter)
         else:
             dc_evse_charge_params = (
                 await self.comm_session.evse_controller.get_dc_evse_charge_parameter()
             )
-            ev_max_voltage = (
-                charge_params_req.dc_ev_charge_parameter.ev_maximum_voltage_limit
-            )
-            ev_max_current = (
-                charge_params_req.dc_ev_charge_parameter.ev_maximum_current_limit
-            )
-            ev_energy_request = (
-                charge_params_req.dc_ev_charge_parameter.ev_energy_request
-            )
-            ev_max_power = (
-                charge_params_req.dc_ev_charge_parameter.ev_maximum_power_limit
-            )
-            ev_charge_params_limits = EVChargeParamsLimits(
-                ev_max_voltage=ev_max_voltage,
-                ev_max_current=ev_max_current,
-                ev_max_power=ev_max_power,
-                ev_energy_request=ev_energy_request,
-            )
-            departure_time = charge_params_req.dc_ev_charge_parameter.departure_time
-
-        self.comm_session.evse_controller.ev_charge_params_limits = (
-            ev_charge_params_limits
-        )
+            ev_data_context.update_dc_charge_parameters(charge_params_req.dc_ev_charge_parameter)
 
         if not departure_time:
             departure_time = 0
```

Comment:
```suggestion
        departure_time = (
            ev_data_context.departure_time if ev_data_context.departure_time else 0
        )
```

Reply:
it was updated, closing for a fresh review